File: MAT3007/Python/Assignment 6/Q1.py
# -*- coding: utf-8 -*-
"""
Created on Sat Apr  9 21:08:10 2022

@author: Mark
"""

import numpy as np
from mpl_toolkits.mplot3d import Axes3D
from matplotlib import pyplot as plt

fig = plt.figure()
ax = Axes3D(fig)
x = np.arange(-5,5,0.1)
y = np.arange(-5,5,0.1)
x, y = np.meshgrid(x,y)
z = x**4 + 2/3 * (x**3) + 1/2 * (x**2) - 2*(x**2)*y + 4/3 *(y**2)
surf = ax.plot_surface(x,y,z,rstride = 1, cstride = 1, cmap = 'rainbow')
fig.colorbar(surf, shrink = 0.5, aspect = 5)
plt.show()

File: MAT3007/MATLAB/Assignment 6/Q1.m
x = linspace(-5, 5, 100);
y = linspace(-5, 5, 100);
[X, Y] = meshgrid(x, y);
Z = X + 2/3 * X.^3 + 1/2 * X.^2 - 2 * Y .* X.^2 + 4/3 * Y.^2;
figure;
surf(X, Y, Z);
hold on;
plot3(0, 0, 0, 'rx')
plot3(-1, 3/4, 1/12, 'go')
colorbar();
xlabel('x1');
ylabel('x2');
hold off;
